### PR TITLE
Replace more assigns with tokens in invoice, add tests

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
https://github.com/civicrm/civicrm-core/pull/23816 +  a few more swaps

Technical Details
----------------------------------------
This is part of working through the fixmes in https://github.com/civicrm/civicrm-core/compare/master...totten:cont_work?expand=1 to get to the point where all variables used in message templates are predictable with tokens used where possible

Comments
----------------------------------------

### CODE REVIEW

**Comment by mlutfy** [comment]
I would be comfortable merging this based on code-review & tests, but will let someone else merge because of lack of sleep.

File: xml/templates/message_templates/contribution_invoice_receipt_html.tpl [hunk 1]
```diff
@@ -161,7 +161,7 @@
         </tr>
       </table>
 
-      {if $contribution_status_id == $pendingStatusId && $is_pay_later == 1}
+      {if '{contribution.contribution_status_id:name}' == 'Pending' && '{contribution.is_pay_later'} == 1}
```
Comment (demeritcowboy) [inline xml/templates/message_templates/contribution_invoice_receipt_html.tpl:164]: Also.

Comment (eileenmcnaughton) [reply xml/templates/message_templates/contribution_invoice_receipt_html.tpl:164]: surely I didn't copy & paste

Comment (eileenmcnaughton) [reply xml/templates/message_templates/contribution_invoice_receipt_html.tpl]: fixed

File: xml/templates/message_templates/contribution_invoice_receipt_html.tpl [hunk 2]
```diff
@@ -150,7 +150,7 @@
               <tr>
                 <td colspan="5"></td>
               </tr>
-              {if $contribution_status_id == $pendingStatusId && $is_pay_later == 1}
+              {if '{contribution.contribution_status_id:name}' == 'Pending' && '{contribution.is_pay_later'} == 1}
```
[the remark this answers is not in the document]
Comment (eileenmcnaughton) [reply xml/templates/message_templates/contribution_invoice_receipt_html.tpl]: fixed

**Comment by eileenmcnaughton** [comment]
thanks @demeritcowboy

**Pull request merged**